### PR TITLE
Document ConfigMap Key Ref support

### LINKED ISSUE #3341: Document configmap key ref
relates to: https://github.com/spring-cloud/spring-cloud-deployer-kubernetes/issues/298

depends on: https://github.com/spring-cloud/spring-cloud-deployer-kubernetes/issues/298

### PR DESCRIPTION
**Pull request by chrisjs**
Resolves #3341

### CODE REVIEW

File: spring-cloud-dataflow-docs/src/main/asciidoc/getting-started-kubernetes.adoc [hunk 1]
```diff
@@ -698,3 +698,67 @@ data:
...
+----
+deployer.<app>.kubernetes.configMapKeyRefs=[{envVarName: 'MY_CM', configMapName: 'testcm', dataKey: 'platform'}]
+----
+====
+
+Replace `<app>` with the name of your application and the `envVarName`, `configMapName` and `dataKey` attributes with the appropriate values for your application environment and ConfigMap.
+
+You can configure ConfigMap references at the global server level as well.
+
+The following example shows how to do so for streams:
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-docs/src/main/asciidoc/getting-started-kubernetes.adoc]: Perhaps point/link to the skipper deployment YAML file for reference?

File: spring-cloud-dataflow-docs/src/main/asciidoc/getting-started-kubernetes.adoc [hunk 2]
```diff
@@ -698,3 +698,67 @@ data:
...
+                accounts:
+                  default:
+                    configMapKeyRefs:
+                      - envVarName: MY_CM
+                        configMapName: testcm
+                        dataKey: platform
+----
+====
+
+The following example shows how to do so for tasks:
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-docs/src/main/asciidoc/getting-started-kubernetes.adoc]: Perhaps point/link to the scdf deployment YAML file for reference?

**Comment by chrisjs** [comment]
it should be the same generic copy/paste as the others. keeping things generic lessens the burden of updating more in the future. if you want hardcoded paths ill update, but..

File: spring-cloud-dataflow-docs/src/main/asciidoc/getting-started-kubernetes.adoc (hunk 2 above)
Comment (sabbyanandan) [reply spring-cloud-dataflow-docs/src/main/asciidoc/getting-started-kubernetes.adoc]: > it should be the same generic copy/paste as the others. keeping things generic lessens the burden of updating more in the future. if you want hardcoded paths ill update, but..

Not about the path to the file really, but stating something like: "change the `server-deployment.yaml` with the following", might help.

Comment (chrisjs) [reply spring-cloud-dataflow-docs/src/main/asciidoc/getting-started-kubernetes.adoc]: :+1:

**Comment by ilayaperumalg** [comment]
LGTM, verified and merging.

**Pull request merged**